### PR TITLE
add TMHeader check for DepositTxCallback

### PR DESCRIPTION
**Pull request by hoangdv2429**
## 1. Summary
Fixes # (issue)

This PR add missing TMHeader check in DepositTxCallback.

## 2.Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 4. How to test/use

The test for this is currently ongoing

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Comment by joe-bowman** [comment]
I believe this should be targetting main?

**Comment by joe-bowman** [comment]
Otherwise, lgtm

**Comment by hoangdv2429** [comment]
> I believe this should be targetting main?

I changed target branch, thank you!

File: x/interchainstaking/keeper/callbacks.go [hunk 1]
```diff
@@ -454,6 +454,12 @@ func DepositTxCallback(k *Keeper, ctx sdk.Context, args []byte, query icqtypes.Q
 		return err
 	}
 
+	// check client state validity
+	err = k.CheckTMHeaderForZone(ctx, &zone, res)
+	if err != nil {
+		return err
+	}
```
Comment (ThanhNhann) [inline x/interchainstaking/keeper/callbacks.go:461]: I think we should move these lines to start from the line `451` after we call `keeper.GetReceipt`

**Review by ThanhNhann — COMMENTED** [review]
Let's `lint` it @hoangdv2429

**Comment by hoangdv2429** [comment]
@joe-bowman @ThanhNhann I addressed all issue

**Review by ThanhNhann — APPROVED** [review]

**Review by joe-bowman — APPROVED** [review]
lgtm, thank you

**Pull request merged**